Assets/Scripts/PlayerInput/KeyboardMouseInput.cs

```diff
@@ -0,0 +1,194 @@
+        }
+
+        /// <summary>
+        /// Control logic for Direct control mode
+        /// </summary>
+        private void ResolveControlBot()
+        {
+            //Switch to selection mode when corresponding key is pressed
+            if (Input.GetKeyDown(GoToSelectModeKey))
+            {
+                selectedBot = currentBot;
+                currentBot = null;
+                SelectionModeActivated?.Invoke(this, EventArgs.Empty);
+                return;
+            }
+
+            //Orient bot towards the mouse
+            AlignRotationWithTheMouse();
+            //Move bot accordingly to the player's input
+            Vector3 movementDirection = new Vector3(Input.GetAxis(InputDictionary.Horizontal), 0f, Input.GetAxis(InputDictionary.Vertical)).normalized;
+            currentBot.MoveTowards(movementDirection);
+
+            if(Input.GetButtonDown(InputDictionary.Jump))
+            {
+                currentBot.PerformDash(movementDirection);
+            }
+            if(Input.GetButtonDown(InputDictionary.Attack))
+            {
+                currentBot.UseHeldItem();
+            }
+            if(Input.GetButtonDown(InputDictionary.Dig))
+            {
+                currentBot.TryToDig();
+            }
+        }
+        /// <summary>
+        /// Control logic for Selection mode.
+        /// </summary>
+        private void ResolveChoosingBot()
+        {
+            //Get all the bots that are still alive
+            var bots = Gamemaster.Instance.GetAIMBots().Where(bot => bot.Alive);
+            if(!bots.Any())
+            {
+                //All bots are dead! Nothing to do here
+                return;
+            }
+            //Select the first bot available if no bot is selected or previously selected is no longer available
+            if (selectedBot == null || !bots.Contains(selectedBot))
+            {
+                selectedBot = bots.FirstOrDefault();
+                SelectionChanged?.Invoke(this, new AIMBotEventArgs(selectedBot));
+            }
+            //Switch to the direct control when the corresponding key is pressed
+            if (Input.GetKeyDown(ConfirmSelectionKey))
+            {
+                currentBot = selectedBot;
+                selectedBot = null;
+                timeSinceLastSelection = 0f;
+                ControlModeActivated?.Invoke(this, new AIMBotEventArgs(currentBot));
+                return;
+            }
+            //Timeout between jumps
+            if (!Mathf.Approximately(timeSinceLastSelection, 0))
+            {
+                timeSinceLastSelection -= Time.deltaTime;
+                if (timeSinceLastSelection < 0f)
+                {
+                    timeSinceLastSelection = 0f;
+                }
+                else
+                {
+                    return;
+                }
+            }
+
+            var choosingDirection = new Vector3(Input.GetAxis("Horizontal"), 0f, Input.GetAxis("Vertical")).normalized;
```

Comment:
Could use your dictionary here
```suggestion
            var choosingDirection = new Vector3(Input.GetAxis(InputDictionary.Horizontal), 0f, Input.GetAxis(InputDictionary.Vertical)).normalized;
```